components/automate-ui/src/app/services/chef-session/chef-session.service.ts

```diff
@@ -61,36 +70,45 @@ export class ChefSessionService implements CanActivate {
       // and components/session-service and chef-session-service -- it would
       // make sense to try to refresh the session if we get a 401 from the API,
       // before giving up and calling logout().
-      window.setInterval(() => {
-        /* Using XMLHttpRequest instead of angular/http to avoid
-           dependency hell. */
-        const xhr = new XMLHttpRequest();
-        xhr.onreadystatechange = this.refreshSessionCallback.bind(this);
-        xhr.open('GET', '/session/refresh', true);
-        xhr.responseType = 'json';
-        xhr.setRequestHeader('Authorization', `Bearer ${this.id_token}`);
-        xhr.send(null);
-      }, minute);
+      timer(0, minute).pipe(
```

Comment:
OK, question for you experts: using `window.setInterval` was one of the reasons that we couldn't have our automate-ui e2e protractor tests wait for the angular state to "settle". Does the same thing apply to `timer`? (I think the workaround would have been to cram this into "angular's zone" somehow, but it's really beyond my understanding of this.)

Reply:
dont know the answer to that question, but i've somehow broken the cypress tests :(